server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts

```diff
@@ -505,22 +507,59 @@ export class SummaryWriter implements ISummaryWriter {
     private async generateLogtailEntries(
         from: number,
         to: number,
-        pending: ISequencedOperationMessage[]): Promise<ITreeEntry[]> {
+        pending: ISequencedOperationMessage[],
+        lastSummaryMessages: ISequencedDocumentMessage[] | undefined): Promise<ITreeEntry[]> {
         const logTail = await this.getLogTail(from, to, pending);
+
+        // Some ops would be missing if we switch cluster during routing.
+        // We need to load these mssing ops from the last summary.
+        const missingOps = await this.getSummaryOps(from, to, logTail, lastSummaryMessages);
+        const fullLogTail = missingOps ?
+            (missingOps.concat(logTail)).sort((op1, op2) => op1.sequenceNumber - op2.sequenceNumber) :
+            logTail;
+
+        // Check the missing operations in the fullLogTail
+        if (fullLogTail.length !== (to - from - 1)) {
+            const missingOpsSequenceNumbers: number[] = [];
+            const fullLogTailSequenceNumbers = fullLogTail.map((ms) => ms.sequenceNumber);
+            for (let i = from + 1; i < to; i++) {
+                if (!fullLogTailSequenceNumbers.includes(i)) {
+                    missingOpsSequenceNumbers.push(i);
```

Comment:
That is correct but we get into this loop only in the cases where there are missing ops (line 522) which doesn't happen often. But when it does, it will be useful to have this info.

Reply:
I think we don't have to do this because like he said the log tail is not too long, and also it only comes to error cases.